state-sync/state-sync-v2/data-streaming-service/src/streaming_client.rs

```diff
@@ -0,0 +1,343 @@
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum StreamRequest {
+    GetAllAccounts(GetAllAccountsRequest),
+    GetAllEpochEndingLedgerInfos(GetAllEpochEndingLedgerInfosRequest),
+    GetAllTransactions(GetAllTransactionsRequest),
+    GetAllTransactionOutputs(GetAllTransactionOutputsRequest),
+    ContinuouslyStreamTransactions(ContinuouslyStreamTransactionsRequest),
+    ContinuouslyStreamTransactionOutputs(ContinuouslyStreamTransactionOutputsRequest),
+    RefetchNotificationPayload(RefetchNotificationPayloadRequest),
+}
+
+/// A client request for fetching all account states at a specified version.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct GetAllAccountsRequest {
+    pub version: Version,
+}
+
+/// A client request for fetching all available epoch ending ledger infos.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct GetAllEpochEndingLedgerInfosRequest {
+    pub start_epoch: Epoch,
+}
+
+/// A client request for fetching all transactions with proofs.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct GetAllTransactionsRequest {
+    pub start_version: Version,
+    pub end_version: Version,
+    pub max_proof_version: Version,
+    pub include_events: bool,
+}
+
+/// A client request for fetching all transaction outputs with proofs.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct GetAllTransactionOutputsRequest {
+    pub start_version: Version,
+    pub end_version: Version,
+    pub max_proof_version: Version,
+}
+
+/// A client request for continuously streaming transactions with proofs (with no end).
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ContinuouslyStreamTransactionsRequest {
+    pub start_version: Version,
+    pub start_epoch: Epoch,
+    pub include_events: bool,
+}
+
+/// A client request for continuously streaming transaction outputs with proofs (with no end).
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ContinuouslyStreamTransactionOutputsRequest {
+    pub start_version: Version,
+    pub start_epoch: Epoch,
+}
+
+/// A client request for refetching a notification payload.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct RefetchNotificationPayloadRequest {
+    pub notification_id: NotificationId,
+    pub refetch_reason: PayloadRefetchReason,
+}
+
+/// The reason for having to refetch a data payload in a data notification.
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[allow(dead_code)]
+pub enum PayloadRefetchReason {
+    InvalidPayloadData,
+    PayloadTypeIsIncorrect,
+    ProofVerificationFailed,
+}
+
+/// The streaming service client that talks to the streaming service and
+/// implements the `DataSteamingClient` interface.
+pub struct StreamingServiceClient {
+    request_sender: mpsc::UnboundedSender<StreamRequestMessage>,
+}
+
+impl StreamingServiceClient {
+    #[allow(dead_code)]
+    pub fn new(request_sender: mpsc::UnboundedSender<StreamRequestMessage>) -> Self {
+        Self { request_sender }
+    }
+
+    async fn send_request_and_await_response(
```

Comment:
I think just `send_request` is probably enough, given the result is the `DataStreamListener`